scipy/interpolate/_bsplines.py

```diff
@@ -314,6 +322,14 @@ def __call__(self, x, nu=0, extrapolate=None):
         x = np.asarray(x)
         x_shape, x_ndim = x.shape, x.ndim
         x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+
+        # With periodic extrapolation we map x to the segment
+        # [self.t[k], self.t[-k]].
+        if extrapolate == 'periodic':
+            x = self.t[self.k] + (x - self.t[self.k]) % (self.t[-(self.k)] -
```

Comment:
This was the reason `b(xx)` returned some `nan` values since I was taking `t[n+1]` instead of `t[n]` as the upper limit. It should have been `t[-k-1]`.